system/lib/pthread/emscripten_futex_wait.c

```diff
@@ -25,6 +25,7 @@ int emscripten_futex_wait(volatile void *addr, uint32_t val, double max_wait_ms)
   // __builtin_wasm_memory_atomic_wait32 so we call out the JS function that
   // will busy wait.
   if (!_emscripten_thread_supports_atomics_wait()) {
+    emscripten_check_blocking_allowed();
     ret = _emscripten_futex_wait_non_blocking(addr, val, max_wait_ms);
```

Comment:
I'm confused. This line does a non-blocking wait, so why check for blocking before it?

Reply:
The job of `emscripten_check_blocking_allowed` is to report the warning about busy waiting on main (e.g. `Blocking on the main thread is very dangerous..`). 

However, this is normally called before `emscripten_futex_wait` is called.   For example in `pthread_cond_timedwait.c` and `pthread_join.c`.     I guess there is no harm in *also* calling it from here.. but generally I think it should have already been called before we get here.

Reply:
@sbc100 Here is the stack trace that displays the warning with this change, but hasn't done it before.
I'm not familiar enough with the architecture to know where the better place for this call would be...?
![image](https://user-images.githubusercontent.com/72875131/149098520-083efff9-eda4-4b61-840e-51c87bee5738.png)